src/main/java/com/gradle/develocity/bamboo/DevelocityPreJobAction.java

```diff
@@ -56,20 +61,28 @@ public void execute(@NotNull StageExecution stageExecution, @NotNull BuildContex
         String accessKey = credentials.getPassword();
         if (StringUtils.isBlank(accessKey)) {
             LOGGER.warn(
-                "Shared credentials with the name {} do not have password set. Environment variable {} will not be set",
-                sharedCredentialName, Constants.DEVELOCITY_ACCESS_KEY
+                    "Shared credentials with the name {} do not have password set. Environment variable {} will not be set",
+                    sharedCredentialName, Constants.DEVELOCITY_ACCESS_KEY
             );
             return;
         }
 
         injectors.stream()
-            .filter(i -> i.hasSupportedTasks(buildContext))
-            .map(i -> i.buildToolConfiguration(configuration))
-            .filter(BuildToolConfiguration::isEnabled)
-            .findFirst()
-            .ifPresent(__ ->
-                buildContext
-                    .getVariableContext()
-                    .addLocalVariable(Constants.ACCESS_KEY, accessKey));
+                .filter(i -> i.hasSupportedTasks(buildContext))
+                .map(i -> i.buildToolConfiguration(configuration))
+                .filter(BuildToolConfiguration::isEnabled)
+                .findFirst()
+                .flatMap(__ ->
+                        shortLivedTokenClient.get(
+                                configuration.getServer(),
+                                DevelocityAccessKey.of(accessKey),
+                                configuration.getShortLivedTokenExpiry()
+                        )
+                )
+                .ifPresent(shortLivedToken ->
```

Comment:
If we fail to get a short-lived access token, do we want to fail the build, or fall back to some reasonable behavior?

In our own dogfooding case, we decided to not set the access key and let the build fail to use various Develocity features like build caching and build scan publishing. However, we had to explicitly disable Test Distribution (which will otherwise fail if it fails to connect to Develocity).  I wonder if we should do something similar here.

I also wonder if it would make sense to add a configuration option to allow users to decide if they want the build to fail if it can't get an access token, or to continue without Develocity.